tests/TESTINS5.MLC

```diff
@@ -341,6 +373,146 @@ BIC_OK2  EQU    *
          IILL   R3,4
          CLG    R3,=AL2(1,2,3,4)
          RT1    CC0
+* MGRK
+         LMG   R5,R6,=FL8'2,3'
+         MGRK  R2,R5,R6
+         STMG  R2,R3,DWORK
+         CLC   DWORK,=FL16'6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'2,-3'
+         MGRK  R2,R5,R6
+         STMG  R2,R3,DWORK
+         CLC   DWORK,=FL16'-6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'-2,3'
+         MGRK  R2,R5,R6
+         STMG  R2,R3,DWORK
+         CLC   DWORK,=FL16'-6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'-2,-3'
+         MGRK  R2,R5,R6
+         STMG  R2,R3,DWORK
+         CLC   DWORK,=FL16'6'
+         RT1    CC0 
+         LMG   R5,R6,=FL8'123456,10000000'
+         MGRK  R2,R5,R6
+         STMG  R2,R3,DWORK
+         CLC   DWORK,=FL16'1234560000000'
+         RT1    CC0 
+* MS
+         L      R1,=F'2'
+         MS    R1,=F'3'
+         CL     R1,=F'6'
+         RT1  CC0
+         L       R1,=F'2'
+         MS    R1,=F'-3'
+         CL     R1,=F'-6'
+         RT1  CC0
+         L       R1,=F'-2'
+         MS    R1,=F'-3'
+         CL     R1,=F'6'
+         RT1  CC0
+* MSY
+         L      R1,=F'2'
+         MSY    R1,=F'3'
+         CL     R1,=F'6'
+         RT1  CC0
+         L       R1,=F'2'
+         MSY    R1,=F'-3'
+         CL     R1,=F'-6'
+         RT1  CC0
+         L       R1,=F'-2'
+         MSY    R1,=F'-3'
+          CL     R1,=F'6'
+          RT1  CC0
+* MSGRKC
+         LMG   R5,R6,=FL8'2,3'
+         MSGRKC  R2,R5,R6
+         CLG    R2,=FL8'6'
```

Comment:
Since MSGRKC sets the condition code, which would be 2 here, I think, you should first set the condition code to something other than the expected condition code then test it after the instruction executes. Something like

```
         LMG   R5,R6,=FL8'2,3'
         SR    R0,R0      Set condition code to 0
         MSGRKC  R2,R5,R6
         RT1   CC2         Condition code must be 2
         CLG   R2,=FL8'6'
```